### ISSUE TITLE
GridFS uncatched exception when no fs.chunks exists for given fs.file

### ISSUE THREAD

**Issue by jankapunkt** [VeliovGroup/Meteor-Files#730]
- Give an expressive description of what is went wrong

When using GridFS (no matter, whether `gridfs-stream` or the internal `GridFsBucket` I get an uncatched error, when `link` a file for download but the chunks in `fs.chunks` (or the respective bucket name) does not exists.

This causes a complete crash on the server without restart attempt.

Reproduction:

* create a new project, implement the minimal setup from the wiki
* Upload any file
* open mongo shell `meteor mongo`
* delete chunks `db.fs.chunks.remove({})` => should return a number above 0
* try to `link` the file for creating a download => server crash 

(see log at the bottom)

Edit: Expected behavior would be that the error is catched and sent to client within the appropriate HTTP response (I think this could be a 404 as the chunks are not found by given file id but could also be a 500 because this could also be thrown when chunks are corrupted)

  - Version of `Meteor-Files` you're experiencing this issue

1.14.0

  - Version of `Meteor` you're experiencing this issue

1.9.2
  - Where this issue appears? OS (Mac/Win/Linux)? Browser name and its version?

OSX / Ubuntu 18.04.

  - Is it *Client* or *Server* issue?

Server

  - Post *Client* and/or *Server* logs with enabled `debug` option, you can enable "debug" mode in [*Constructor*](https://github.com/VeliovGroup/Meteor-Files/wiki/Constructor)
  
```bash


I20200304-09:22:07.373(1)? [FilesCollection] [download(/cdn/storage/documentFiles/JyGYH7BjkwB8RHKJt/original/JyGYH7BjkwB8RHKJt.docx?download=true, original)]
I20200304-09:22:07.374(1)? read static response Personalbogen_dt_engl.docx
W20200304-09:22:07.385(1)? (STDERR) /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:132
W20200304-09:22:07.387(1)? (STDERR)       throw err;
W20200304-09:22:07.387(1)? (STDERR)       ^
W20200304-09:22:07.387(1)? (STDERR) 
W20200304-09:22:07.387(1)? (STDERR) MongoError: no chunks found for file, possibly corrupt
W20200304-09:22:07.387(1)? (STDERR)     at Function.create (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb-core/lib/error.js:43:12)
W20200304-09:22:07.388(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/gridfs/grid_store.js:782:22
W20200304-09:22:07.388(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/gridfs/grid_store.js:1196:7
W20200304-09:22:07.388(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:410:17
W20200304-09:22:07.388(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:398:11
W20200304-09:22:07.388(1)? (STDERR)     at ClientSession.endSession (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb-core/lib/sessions.js:134:41)
W20200304-09:22:07.393(1)? (STDERR)     at executeCallback (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:395:17)
W20200304-09:22:07.394(1)? (STDERR)     at handleCallback (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:128:55)
W20200304-09:22:07.394(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/operations/collection_ops.js:598:5
W20200304-09:22:07.394(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:410:17
W20200304-09:22:07.395(1)? (STDERR)     at executeCallback (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:402:9)
W20200304-09:22:07.395(1)? (STDERR)     at handleCallback (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/utils.js:128:55)
W20200304-09:22:07.395(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/lib/operations/cursor_ops.js:195:5
W20200304-09:22:07.395(1)? (STDERR)     at handleCallback (/home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb-core/lib/cursor.js:204:5)
W20200304-09:22:07.395(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb-core/lib/cursor.js:426:38
W20200304-09:22:07.395(1)? (STDERR)     at /home/jankapunkt/.meteor/packages/npm-mongo/.3.3.0.yo00pe.z7u0f++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb-core/lib/cursor.js:434:46 {
W20200304-09:22:07.396(1)? (STDERR)   driver: true,
W20200304-09:22:07.396(1)? (STDERR)   name: 'MongoError',
W20200304-09:22:07.396(1)? (STDERR)   [Symbol(mongoErrorContextSymbol)]: {}
W20200304-09:22:07.396(1)? (STDERR) }
=> Exited with code: 1
=> Your application is crashing. Waiting for file change.
```

**Comment by jankapunkt** [comment]
Ok I found the error in the[ wiki page on GridFS ](https://github.com/VeliovGroup/Meteor-Files/wiki/GridFS-Integration)there is this code:

```javascript
interceptDownload(http, image, versionName) {
    const _id = (image.versions[versionName].meta || {}).gridFsFileId;
    if (_id) {
      const readStream = gfs.createReadStream({ _id });
      readStream.on('error', err => { throw err; });
      readStream.pipe(http.response);
    }
    return Boolean(_id); // Serve file from either GridFS or FS if it wasn't uploaded yet
  }
```

where the `throw err` will cause the server crash, because it's outside of a fiber.+

This has already been mentioned here: https://github.com/VeliovGroup/Meteor-Files/issues/616#issuecomment-466793977 but the error has not been reported as an own issue.


Solutions would either to use `Meteor.bindEnvironment` or just `console.error` but I think the best solution would be to update the GridFS documentation to the one from the GridFS bucket demo repo: https://github.com/VeliovGroup/files-gridfs-autoform-example/pull/1

I think it should be good to review the PR and merge it and then I will asap update the wiki for native GridFS integration, also finally resolving https://github.com/VeliovGroup/Meteor-Files/issues/616

**Comment by dr-dimitru** [comment]
Hello @jankapunkt,

Thank you for contribution on #616 
Looking forward PR to the docs

**Comment by jankapunkt** [comment]
This issue can be closed, once #731 is merged. The docs related to #616 will be implemented in a separate PR

**Comment by dr-dimitru** [comment]
Merged, wiki updated:
- Up-to-date: https://github.com/VeliovGroup/Meteor-Files/wiki/GridFS-Bucket-Integration
- Legacy: https://github.com/VeliovGroup/Meteor-Files/wiki/GridFS-Integration
